lib/cns-client/rs/tests/register_and_lookup.rs

```diff
@@ -0,0 +1,165 @@
+    let test_client_wasm =
+        fs::read(TEST_CLIENT_WASM).expect("Wasm file not found, run 'dfx build'.");
+    pic.install_canister(cns_root, cns_root_wasm, vec![], None);
+    pic.install_canister(tld_operator, tld_operator_wasm, vec![], None);
+    pic.install_canister(
+        test_client,
+        test_client_wasm,
+        encode_one(Some(ClientInit {
+            cns_root_cid: cns_root.to_string(),
+        }))
+        .unwrap(),
+        None,
+    );
+    // Set controller, so that test_client can register domains.
+    pic.set_controllers(tld_operator, None, vec![test_client])
+        .expect("Failed setting TLD operator controller");
+    CnsSetup {
+        pic,
+        cns_root,
+        tld_operator,
+        test_client,
+    }
+}
+
+impl CnsSetup {
+    fn register_icp_nc(&self) {
+        let registration_records = RegistrationRecords {
+            controller: vec![],
+            records: Some(vec![DomainRecord {
+                name: ".icp.".to_string(),
+                record_type: "NC".to_string(),
+                ttl: 3600,
+                data: self.tld_operator.to_string(),
+            }]),
+        };
+        self.pic
+            .update_call(
+                self.cns_root,
+                Principal::anonymous(),
+                "register",
+                encode_args((&".icp.", &registration_records)).expect("failed encoding args"),
+            )
+            .expect("Failed registering NC for icp");
+    }
+
+    fn register_domain(&self, domain: &str, cid_text: &str) -> Result<(), CnsError> {
+        let response = self.pic.update_call(
+            self.test_client,
+            Principal::anonymous(),
+            "register_domain",
+            encode_args((&domain, &cid_text)).expect("failed encoding args"),
+        );
+        let Ok(WasmResult::Reply(reply)) = response else {
+            panic!("call failed: {:?}", response);
+        };
+        Decode!(&reply, Result<(), CnsError>).expect("reply decoding failed")
+    }
+
+    fn lookup_domain(&self, domain: &str) -> Result<Principal, CnsError> {
+        let response = self.pic.query_call(
+            self.test_client,
+            Principal::anonymous(),
+            "lookup_domain",
+            encode_one(domain).expect("failed encoding arg"),
+        );
+        let Ok(WasmResult::Reply(reply)) = response else {
+            panic!("call failed: {:?}", response);
+        };
+        Decode!(&reply, Result<Principal, CnsError>).expect("reply decoding failed")
+    }
+}
+
+#[test]
+fn should_register_and_lookup() {
+    let env = cns_setup();
+    env.register_icp_nc();
+    for (domain, cid_text) in [
+        ("example.icp.", "aaaaa-aa"),
+        ("nns_governance.icp.", "rrkah-fqaaa-aaaaa-aaaaq-cai"),
+        ("nns_registry.icp.", "rwlgt-iiaaa-aaaaa-aaaaa-cai"),
+    ] {
+        let result = env.register_domain(domain, cid_text);
+        assert!(result.is_ok(), "Domain registration failed: {:?}", result);
```

Comment:
This is just a little more concise and does the same thing.

```suggestion
        let result = env.register_domain(domain, cid_text).expect(format!("Domain registration failed: {:?}", result)));
```

Reply:
I believe the suggestions would not work, as we don't have `result` to print in `format!`.